scipy/stats/_continuous_distns.py

```diff
@@ -10513,20 +10513,24 @@ def _pdf(self, x, b, c):
 
     def _logpdf(self, x, b, c):
         b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
-        pdf = b * x**-(b+1) / (1 - 1/c**b)
-        return np.log(pdf)
+        return xpx.apply_where(b > 0, (x, b, c), self._logpdf_pos_b, super()._logpdf)
```

Comment:
Using `super()._logpdf` is not ideal here, but it's easier than figuring out the best logspace implementation, and it's more compact and actually better (due to logic in `super()._logpdf`) than `lambda x, b, c: np.log(self._pdf(x, b, c))`.